### PR TITLE
[RLlib] Enable .py files with AlgorithmConfig objects in them to be run by our `run_regression_test.py` script.

### PR DESCRIPTION
**Pull request by sven1977**
Signed-off-by: sven1977 <svenmika1977@gmail.com>

Enable .py files with AlgorithmConfig objects in them to be run by our `run_regression_test.py` script.
* In preparation of converting CI learning tests to .py files (and thus making it easier to configure multi-agent and callbacks  settings).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/tests/run_regression_tests.py [hunk 1]
```diff
@@ -56,43 +60,50 @@
...
-    if os.path.isfile(abs_yaml_path):
-        yaml_files = [abs_yaml_path]
-    # Given path/file does not exist.
-    elif not os.path.isdir(abs_yaml_path):
-        raise ValueError("yaml-dir ({}) not found!".format(args.yaml_dir))
+    if os.path.isfile(abs_path):
+        files = [abs_path]
     # Path given -> Get all yaml files in there via rglob.
+    elif os.path.isdir(abs_path):
+        files = rllib_dir.rglob(args.dir + "/*.yaml")
```
Comment (maxpumperla) [inline rllib/tests/run_regression_tests.py:84]: shouldn't this also look for `py` files? or do we want to make this a follow-up?

File: rllib/tests/run_regression_tests.py [hunk 2]
```diff
@@ -56,43 +60,50 @@
...
 
     # Loop through all collected files.
-    for yaml_file in yaml_files:
-        experiments = yaml.safe_load(open(yaml_file).read())
+    for file in files:
+        # For python files, need to make sure, we only deliver the module name into the
+        # `load_experiments_from_file` function (everything from "/ray/rllib" on).
+        if file.endswith(".py"):
+            file = re.sub("^.*/ray/rllib/", "ray/rllib/", file)
+            experiments = load_experiments_from_file(file, SupportedFileType.python)
```
Comment (maxpumperla) [inline rllib/tests/run_regression_tests.py:100]: awesome!

**Review by maxpumperla — APPROVED** [review]
cool, just a quick question for my understanding!

File: rllib/tests/run_regression_tests.py (hunk 1 above)
Comment (sven1977) [reply rllib/tests/run_regression_tests.py:84]: Oh, great catch! Let me fix this.

Comment (sven1977) [reply rllib/tests/run_regression_tests.py]: Fixed, pending tests ...

**Pull request merged**